Tutorials/Members-Registration-And-Logins/index-v10.md

```diff
@@ -0,0 +1,360 @@
+
+Now that we have the options to:
+
+- Register a member
+- Log in a member
+- Check current login status
+- Log out a member
+
+We can also go a bit further and specify which parts of our website should be accessible to logged in members.
+To do this, head on over to the Member section in the Backoffice, and create a new Member Group.
+
+![New member group](images/v8-12-New-Member-Group.png)
+
+Give the group a name, and save it. Next, move over to the Members folder, click on any of the Members you  have, and assign the newly created Member Group to them. In order to do that, click on the member who you would like to assign to the group.
+Under `Properties`, you have the option to pick a group to assign the member to `(Member Group` property).
+
+![Member group assign](images/v8-13-Member-Group-Assign.png)
+
+Almost there. Save the member, move to the Content section and find the page you would like to restrict. Right-click on that content node, and find the option "Restrict Public Access".
+
+![Restrict public access](images/v8-14-Restrict-Public-Access.png)
+
+You will be able to restrict access to a specific member, or a specific group. Choose the latter option.
+On this menu, you will be able to select the group that will have access to the page, the page with the login form, and the page that would be displayed if selected content would be inaccessible to the chosen group.
+
+:::note
+To restrict access you need to have made a member group, and a page containing the login partial. Having a "No access"/error page is recommended, as well - though you can use any content page you have.
+:::
+
+![Restrict public access details](images/v8-15-Restrict-Public-Access-Details.png)
+
+Congratulations! With all of that set up, the page you chose will redirect the user to the Login page if they are not logged in, and if they are logged in - they will be able to see the page's contents.
+
+In the navigation, you will be able to tell the content node is restricted - the small red icon that is added on top of the document type icon signifies that.
+
+![Protected content](images/v8-16-Protected-node.png)
+
+:::note
+The above approach relies on two Macro partial views and a non-Macro partial view. It is also possible to achieve the same result by working entirely with plain Partial Views, three Macros, and even plain HTML/Razor code copied from the Snippets into your Templates.
+:::
+
+However, with the above approach, members will not be assigned to any group automatically - for this to happen, we would need to write a bit of custom code.
+
+## Assigning new members to groups automatically¨
+
+### Creating a new SurfaceController
+
+Since the member saving form is processed in a controller, we can copy the default [UmbRegisterController](https://github.com/umbraco/Umbraco-CMS/blob/v10/contrib/src/Umbraco.Web.Website/Controllers/UmbRegisterController.cs) and add a function to assign the newly created member to a group.
```

Comment:
```suggestion
:::note
To restrict access you need to have made a member group, and a page containing the login partial. Having a "No access"/error page is recommended, as well - though you can use any content page you have.
:::

![Restrict public access details](images/v8-15-Restrict-Public-Access-Details.png)

Congratulations! With all of that setup, the page you chose will redirect the user to the Login page if they are not logged in, and if they are logged in - they will be able to see the page's contents.

In the navigation, you will be able to tell the content node is restricted - the small red icon that is added on top of the document type icon signifies that.

![Protected content](images/v8-16-Protected-node.png)

:::note
The above approach relies on two Macro partial views and a non-macro partial view. It is also possible to achieve the same result by working entirely with plain Partial Views, three Macros, and even plain HTML/Razor code copied from the Snippets into your Templates.
:::

However, with the above approach, members will not be assigned to any group automatically - for this to happen, we would need to write a bit of custom code.

## Assigning new members to groups automatically¨

### Creating a new SurfaceController

Since the member saving form is processed in a controller, we can copy the default [UmbRegisterController](https://github.com/umbraco/Umbraco-CMS/blob/v10/contrib/src/Umbraco.Web.Website/Controllers/UmbRegisterController.cs) and add a function to assign the newly created member to a group.
```